sequencer/src/auction.rs

```diff
@@ -0,0 +1,103 @@
+use crate::{state::FeeAmount, NamespaceId, Transaction};
+use hotshot_types::data::ViewNumber;
+use std::num::NonZeroU64;
+
+/// State Machine for Auction
+#[derive(Clone, Copy)]
+struct Auction {
+    /// current phase.
+    phase: AuctionPhase,
+    /// A structure to find which Phase should be the current one.
+    // This could probably be an enum with some restructuring.
+    possible_phases: [AuctionPhase; 3],
+}
+
+impl Auction {
+    fn update(mut self, view: ViewNumber) {
+        self.phase = self
+            .possible_phases
+            .into_iter()
+            .find(|phase| (view > phase.start && view < phase.end))
+            .unwrap();
+    }
+    fn recv_bid(&self, view: ViewNumber, bid: BidTx) {
+        self.update(view);
+        match self.phase.kind {
+            AuctionPhaseKind::Bid => self.send_txn(bid.as_txn()),
```

Comment:
Rereading, I think the actions that will actually occur are validation and persistence.